### DISCUSSION TITLE
Adding nodes to query connection type

### DISCUSSION THREAD

**Discussion by BengtHagemeister** [hayes/pothos#521] [Q&A] [answered]
Hi,

is there a possibility to get nodes exposed on the connection type?

Something like this but then automatic and including pagination:

```typescript

builder.queryField('posts', (t) =>
    t.prismaConnection(
      {
        type: 'Post',
        cursor: 'id',
        resolve: async (query) =>
          db.posts.findMany({
            ...query,
          }),
      },
      {
        fields: (t) => ({
          nodes: t.prismaField({
            type: ['Post'],
            resolve: async (query) =>
              db.posts.findMany({
                ...query,
              }),
          }),
        }),
      }
    )
  );
```

Following a similar approach as Shopify for example (https://shopify.dev/api/admin-graphql/2022-07/queries/orders#examples-Get_the_first_10_orders).

**Comment by hayes** [comment, marked as answer]
I think you should be able to do nodes: t.field({ type: Post, resolve: (connection) => connection.edges.map(edge => edge.node) })

**Reply by BengtHagemeister** [reply]
Thank you very much! I got it to work with the t.prismaField({}) since I needed the type from there. So the working example including prisma is:

```typescript
builder.queryField('posts', (t) =>
    t.prismaConnection(
      {
        type: 'Post',
        cursor: 'id',
        resolve: async (query) =>
          db.posts.findMany({
            ...query,
          }),
      },
      {
        fields: (t) => ({
          nodes: t.prismaField({
            type: ['Post'],
            resolve: (_query, parent) => parent.edges.map((edge) => edge.node),
          }),
        }),
      }
    )
  );
```

**Reply by hayes** [reply]
I think it's fine in this case, but generally you should avoid using t.prismaField without using/spreading the query parameter.  You can use the return value from builder.prismaObject in the type parameter of other field methods.

**Reply by hayes** [reply]
just added a new `nodesOnConnection` option that will do this automatically

**Reply by BengtHagemeister** [reply]
Awesome thank you!
